src/Codec/Xlsx/StyleSheet.hs

```diff
@@ -0,0 +1,1123 @@
+{-# LANGUAGE CPP                #-}
+{-# LANGUAGE OverloadedStrings  #-}
+{-# LANGUAGE RecordWildCards    #-}
+{-# LANGUAGE TemplateHaskell    #-}
+{-# OPTIONS_GHC -Wall #-}
+-- | Support for writing (but not reading) style sheets
+module Codec.Xlsx.StyleSheet (
+    -- * The main two types
+    StyleSheet(..)
+  , CellXf(..)
+  , minimalStyleSheet
+    -- * Supporting record types
+  , Alignment(..)
+  , Border(..)
+  , BorderStyle(..)
+  , Color(..)
+  , Fill(..)
+  , FillPattern(..)
+  , Font(..)
+  , Protection(..)
+    -- * Supporting enumerations
+  , CellHorizontalAlignment(..)
+  , CellVerticalAlignment(..)
+  , FontFamily(..)
+  , FontScheme(..)
+  , FontVerticalAlignment(..)
+  , LineStyle(..)
+  , PatternType(..)
+  , ReadingOrder(..)
+    -- * Lenses
+    -- ** StyleSheet
+  , styleSheetBorders
+  , styleSheetFonts
+  , styleSheetFills
+  , styleSheetCellXfs
+    -- ** CellXf
+  , cellXfApplyAlignment
+  , cellXfApplyBorder
+  , cellXfApplyFill
+  , cellXfApplyFont
+  , cellXfApplyNumberFormat
+  , cellXfApplyProtection
+  , cellXfBorderId
+  , cellXfFillId
+  , cellXfFontId
+  , cellXfNumFmtId
+  , cellXfPivotButton
+  , cellXfQuotePrefix
+  , cellXfId
+  , cellXfAlignment
+  , cellXfProtection
+    -- ** Alignment
+  , alignmentHorizontal
+  , alignmentIndent
+  , alignmentJustifyLastLine
+  , alignmentReadingOrder
+  , alignmentRelativeIndent
+  , alignmentShrinkToFit
+  , alignmentTextRotation
+  , alignmentVertical
+  , alignmentWrapText
+    -- ** Border
+  , borderDiagonalDown
+  , borderDiagonalUp
+  , borderOutline
+  , borderBottom
+  , borderDiagonal
+  , borderEnd
+  , borderHorizontal
+  , borderStart
+  , borderTop
+  , borderVertical
+  , borderLeft
+  , borderRight
+    -- ** BorderStyle
+  , borderStyleColor
+  , borderStyleLine
+    -- ** Color
+  , colorAutomatic
+  , colorARGB
+  , colorTheme
+  , colorTint
+    -- ** Fill
+  , fillPattern
+    -- ** FillPattern
+  , fillPatternBgColor
+  , fillPatternFgColor
+  , fillPatternType
+    -- ** Font
+  , fontBold
+  , fontCharset
+  , fontColor
+  , fontCondense
+  , fontExtend
+  , fontFamily
+  , fontItalic
+  , fontName
+  , fontOutline
+  , fontScheme
+  , fontShadow
+  , fontStrikeThrough
+  , fontSize
+  , fontUnderline
+  , fontVertAlign
+    -- ** Protection
+  , protectionHidden
+  , protectionLocked
+  ) where
+
+import Control.Lens hiding ((.=))
+import Data.Default
+import Data.Maybe (catMaybes)
+import Text.XML
+import qualified Data.Map as Map
+
+#if !MIN_VERSION_base(4,8,0)
+import Control.Applicative
+#endif
+
+import Codec.Xlsx.Writer.Internal
+
+{-------------------------------------------------------------------------------
+  The main types
+-------------------------------------------------------------------------------}
+
+-- | StyleSheet for an XML document
+--
+-- Relevant parts of the EMCA standard (2nd edition, part 1,
```

Comment:
I sadly don't have time to change all the references in the code, or to test everything again with all versions if we change how we set things up. This works and it's conform the spec (even if it's not the latest). If you want to change it, of course it's your package, and I'd be happy to test it again once you've done that, but unfortunately I won't be able to dedicate the time.